mirgecom/gas_model.py

```diff
@@ -396,25 +478,33 @@ def project_fluid_state(dcoll, src, tgt, state, gas_model, limiter_func=None):
         Thermally consistent fluid state
     """
     cv_sd = op.project(dcoll, src, tgt, state.cv)
+
     temperature_seed = None
     if state.is_mixture:
         temperature_seed = op.project(dcoll, src, tgt, state.dv.temperature)
 
     smoothness_mu = None
     if state.dv.smoothness_mu is not None:
         smoothness_mu = op.project(dcoll, src, tgt, state.dv.smoothness_mu)
+
     smoothness_kappa = None
     if state.dv.smoothness_kappa is not None:
         smoothness_kappa = op.project(dcoll, src, tgt, state.dv.smoothness_kappa)
+
     smoothness_beta = None
     if state.dv.smoothness_beta is not None:
         smoothness_beta = op.project(dcoll, src, tgt, state.dv.smoothness_beta)
 
+    wall_density = None
+    if gas_model.wall is not None:
+        wall_density = op.project(dcoll, src, tgt, state.dv.wall_density)
+
     return make_fluid_state(cv=cv_sd, gas_model=gas_model,
                             temperature_seed=temperature_seed,
                             smoothness_mu=smoothness_mu,
                             smoothness_kappa=smoothness_kappa,
                             smoothness_beta=smoothness_beta,
+                            wall_density=wall_density,
```

Comment:
Per our discussion, it's probably OK to leave this as is until we figure out what will actually be needed by the fluid operator in the end. I'll leave this unresolved though so Mike has a chance to see it.

Reply:
I also have that feeling, and a very similar one about the smoothnii, too.  I understand how/why things got this way as it is easier to piggyback stuff onto existing structure, rather than to duplicate a collection of machinery.   Worth it to make it right now?  Not sure about that.   I struggled with this exact thing a little bit with both this, and with the smoothness stuff.  Ultimately, more important issues drew me away and this seemed to be working properly for our use cases.

Reply:
Issue in #930